app/helpers/repository_helper.rb

```diff
@@ -169,19 +169,19 @@ def remove_folders(folders, user, repo, path: '/', txn: nil)
         if obj.is_a? Repository::RevisionFile
           files << path
         else
-          dirs[path] = current_revision
+          dirs << path
         end
       end
-      dirs[folder_path] = current_revision
+      dirs << folder_path
     end
-    success, file_messages = remove_files(files, user, repo, path: '', txn: txn)
+    success, file_messages = remove_files(files.reverse, user, repo, path: '', txn: txn, option: 'Folder')
```

Comment:
Why is the order that files are deleted in important? Why reverse the order of the list of files here?